### ISSUE TITLE
mvn validate hangs on multi-module project

### ISSUE THREAD

**Issue by beirtipol** [jgitver/jgitver-maven-plugin#130]
### Before submitting an issue I have first:

- [x ] searched for similar already existing issue
- [x ] read the documentation and [wiki](https://github.com/jgitver/jgitver-maven-plugin/wiki) 

*(if you have performed all the above, remove the paragraph and continue describing the issue with template below)*

## Issue

We know that your time is precious, we are thankful that you take time to report us an issue, to make it easier for you please use the following prefilled paragraphs   

**version**: 1.4.3

**usage context**:

- [ ] maven command line: 3.5.2  

**Problem description**:

I have a number of projects I've migrated from SVN to GIT. All my smaller projects work as expected with jgitver. However, our largest and oldest multi-module project hangs immediately when running 'mvn validate'. There is no jgitver.config.xml specified yet, just the extensions file.

What is different about this project is that it has ~30 modules and approx 5 years of log history. There are ~360 tags in the repository which were created by standard maven release builds in SVN. 
 Other projects which have been mvn release'd through SVN work fine so I am guessing it is just the number of submodules which makes this different. 

I have run 'mvn -X validate' to see if anything useful is printed but it seems to stall just after accessing the git config. I'm guessing stuck in a loop before the expected 'jgitver calculated version number' that I see in other projects.

```
[DEBUG] using jgitver on directory: C:\dev\source\bc-calypso-parent\faatgit-clean
[DEBUG] using jgitver on directory: C:\dev\source\bc-calypso-parent\faatgit-clean
[DEBUG] trying to load configuration from: C:\dev\source\bc-calypso-parent\faatgit-clean\.mvn\jgitver.config.xml
[DEBUG] jgitver configuration file C:\dev\source\bc-calypso-parent\faatgit-clean\.mvn\jgitver.config.xml does not exists, skipping it
[DEBUG] trying to load configuration from: C:\dev\source\bc-calypso-parent\faatgit-clean\.mvn\jgtiver.config.xml
[DEBUG] jgitver configuration file C:\dev\source\bc-calypso-parent\faatgit-clean\.mvn\jgtiver.config.xml does not exists, skipping it
[INFO] no suitable configuration file found, using defaults
[DEBUG] readpipe [git, --version],C:\Program Files\Git\cmd
[DEBUG] readpipe may return 'git version 2.14.1.windows.1'
[DEBUG] remaining output:

[DEBUG] readpipe [git, config, --system, --edit],C:\Program Files\Git\cmd
[DEBUG] readpipe may return 'C:/Program Files/Git/mingw64/etc/gitconfig'
[DEBUG] remaining output:

```

I'm not sure where to look next as all other projects work fine:/

**Comment by McFoggy** [comment]
Do you have a special configuration jgitver configuration or the default one?

**Comment by beirtipol** [comment]
It's using default. I haven't added any jgitver config file to the .mvn folder in my project. That's consistent across all projects

**Comment by McFoggy** [comment]
Can you please 
* provide the output of `git describe`
* provide the layout of your history 
`git log --no-color --graph --pretty=format:"%h -%d" --abbrev-commit > history.txt`
* provide the total number of tags `git tag -l | wc -l`  (or just count the lines if you are on windows and do not have `wc` )
* try with a `jgitver.config.xml` file containing 
```xml
<configuration>
    <maxSearchDepth>10</maxSearchDepth>
</configuration>
``` 
and run again `mvn validate`

**Comment by beirtipol** [comment]
- git describe
```
PS C:\dev\source\bc-calypso-parent\faatgit> git describe
fatal: No annotated tags can describe '402b1256610a4f07b45b0abc2dee5b2d1cfd0f31'.
However, there were unannotated tags: try --tags.
PS C:\dev\source\bc-calypso-parent\faatgit> git describe --tags
bc-calypso-parent-2020.04.18.1-81-g402b12566
```

- git history
I'm prevented from attaching files to github from work. The history tree is generally pretty flat. The deepest it goes is recently to:
`| | * | | | | 4926932e2 -`

If you need the file, I'll have a chat with our cyber guys and see if I can get it attached here.

- git tags
372 lines

- adding that config file produces this error:
```
PS C:\dev\source\bc-calypso-parent\faatgit-clean> mvn validate
[ERROR] failure reading configuration from: C:\dev\source\bc-calypso-parent\faatgit-clean\.mvn\jgitver.config.xml: Element 'maxSearchDepth' does not have a match in class fr.brouillard.oss.jgitver.cfg.Configuration at line 2 -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MavenExecutionException
PS C:\dev\source\bc-calypso-parent\faatgit-clean>
```

That's strange. To confirm, here is the content of my extensions.xml

```
<extensions xmlns="http://maven.apache.org/EXTENSIONS/1.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
            xsi:schemaLocation="http://maven.apache.org/EXTENSIONS/1.0.0 http://maven.apache.org/xsd/core-extensions-1.0.0.xsd">
    <extension>
        <groupId>fr.brouillard.oss</groupId>
        <artifactId>jgitver-maven-plugin</artifactId>
        <version>1.4.3</version>
    </extension>
</extensions>
```

**Comment by McFoggy** [comment]
@beirtipol 

what is the version you would expect jgitver to compute for you ? something looking like `2020.04.18.1` ?

* `... No annotated tags can describe...` that mean that probably jgitver is currently not very suited for you project tags
  * if coming from standard maven+SVN+release plugin, then no tag is accessible to jgitver for computation
* the history file generated from previous command has no information in it (comments are striped out), just SHA1, tags and a tree structure ; having a look at it we could just brought an idea of what jgitver should have computed.
* can you use latest `1.5.1` version please?

**Comment by beirtipol** [comment]
Bumping up to 1.5.1 fixed that issue, the config file wasn't relevant. Not sure why I was using 1.4.3, I guess it was from one of the readme.md's I should have checked for a later version.

FYI, I'm not expecting jgitver to figure out versions from our existing tags. I'm aiming to start fresh now that I've moved over to git and aiming to get the versions in line.

I've encountered a different issue now. I'll dig in more and raise a new issue if I can't figure it out. Thanks.
